libs/partners/anthropic/langchain_anthropic/chat_models.py

```diff
@@ -658,6 +666,18 @@ def post_init(cls, values: Dict) -> Dict:
         values["_async_client"] = anthropic.AsyncClient(**client_params)
         return values
 
+    @property
+    def _messages_client(self) -> Messages:
+        if self.beta:
+            return self._client.beta.prompt_caching.messages  # type: ignore[attr-defined]
```

Comment:
this feels more specific than just a "beta" flag indicates. are we going to update client to `beta.{x}.messages` every time there's a new beta feature?

also is cache usage not returned if you use the regular client with the beta headers?

Reply:
Oh, nice, it actually works. here's an example:
```python
from langchain_anthropic import ChatAnthropic
from langchain_core.messages import HumanMessage, SystemMessage

model = ChatAnthropic(
    model="claude-3-opus-20240229",
    temperature=0,
    extra_headers={"anthropic-beta": "prompt-caching-2024-07-31"}
)

chat = [
    SystemMessage([{
        "type": "text",
        "text": "foo"*1000,
        "cache_control": {"type": "ephemeral"},
    }]),
    HumanMessage("Hi"),
]

model.invoke(chat)
```

returning
```
AIMessage(content='Hello! How can I assist you today?', response_metadata={'id': 'msg_01EuihUPN9JrbzZXuZd6oEu8', 'model': 'claude-3-opus-20240229', 'stop_reason': 'end_turn', 'stop_sequence': None, 'usage': {'input_tokens': 8, 'output_tokens': 12, 'cache_creation_input_tokens': 1500, 'cache_read_input_tokens': 0}}, id='run-a13ecd02-d669-4028-b8a2-56e5113d2417-0', usage_metadata={'input_tokens': 8, 'output_tokens': 12, 'total_tokens': 20})
```

Reply:
Is there way to include variables in the system prompt but still includes the "cache_control": {"type": "ephemeral"} parameter?